x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/hooks/use_get_endpoint_exceptions_unavailablle_component.tsx

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type React from 'react';
+import { useUpsellingComponent } from '../../../../../common/hooks/use_upselling';
+
+export const useGetEndpointExceptionsUnavailableComponent = (): React.ComponentType | null => {
+  return useUpsellingComponent('ruleDetailsEndpointExceptions');
+};
```

Comment:
I'd like to see if we can improve code ownership for the changes made to the Rule Details page in this PR.

Can we have somewhere an `endpoint_exceptions` folder that would be owned by you folks? Options off the top of my head:

- `x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/components/endpoint_exceptions`
- `x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/endpoint_exceptions`
- `x-pack/plugins/security_solution/public/detection_engine/endpoint_exceptions/components`

This `endpoint_exceptions` folder would be owned by @elastic/security-defend-workflows and would be exporting components and hooks for use on the Rule Details and optionally other pages of the Detection Engine.

The `useGetEndpointExceptionsUnavailableComponent` hook would live under the `endpoint_exceptions` folder.

Reply:
refactored 35d0a11794d946f21633c0770901b13054bc90ff